plugins/ltac/tacinterp.ml

```diff
@@ -1386,11 +1335,12 @@ and interp_app loc ist fv largs : Val.t Ftactic.t =
         end >>= fun v ->
         (* No errors happened, we propagate the trace *)
         let v = append_trace trace v in
+        Proofview.Goal.enter begin fun gl ->
```

Comment:
You also changed the semantics of Ltac here, by forcing applications to focus. People will start screaming running in circles.

Reply:
Hum, you're raising trouble in my mind. Didn't the `tclORELSE` above already force focussing? Otherwise, this is a real problem, we have a `constr` and we want to print it when in debugging mode and one would have a different behavior in whether we are in debugging or non-debugging mode.

Reply:
No, the `tclORELSE` doesn't force focussing, it's a control operator that leaves goals unchanged. So if you implement the suggestion above that should be OK. Otherwise you can always get the global environment if the goal is not focussed, there is even a tactic primitive `Ftactic.with_env` for that.